src/sagemaker/predictor.py

```diff
@@ -91,10 +94,9 @@ def retrieve_default(
         model_id = inferred_model_id
         model_version = model_version or inferred_model_version or "*"
         inference_component_name = inference_component_name or inferred_inference_component_name
-        config_name = inferred_config_name or None
+        config_name = config_name or inferred_config_name or None
     else:
         model_version = model_version or "*"
-        config_name = None
```

Comment:
i think we still need this here, right?

Reply:
It's defined in the `retrieve_default(... config_name)`, I removed it here because otherwise it will override what's passed in from `retrieve_default`